brave/brave-core
# feat(wallet): add group-by-collection to portfolio NFTs page

Resolves https://github.com/brave/brave-browser/issues/38865



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Create a wallet with multiple NFTs for a collection
2. Open wallet portfolio and click "NFTs"
3. click the filters icon
4. observe the available settings
5. Enable "Group by collection" toggle
6. The NFTs portfolio page should update to display NFTs grouped by collection (may take a few moments to fetch token metadata needed to display the correct collection name)
7. Clicking a collection should navigate to a page that displays all NFTs in that collection

- Search should filter the NFT and NFT collections list
- Saving portfolio filters/settings (accounts, networks) should reset the portfolio NFTs page page-number to "1"
- Reloading the page should initially display the cached NFT collection names until new data is available
- The "Group by collection" toggle should not appear when filtering from the "Assets" tab (only on the NFTs tab)
-
